pkg/selectors/selectors.go

```diff
@@ -236,3 +395,91 @@ func (k *KernelSelectorState) insertAddr6Map(addr6map map[KernelLpmTrie6]struct{
 	k.addr6Maps = append(k.addr6Maps, addr6map)
 	return uint32(mapid)
 }
+
+func (k *KernelSelectorState) createStringMaps() (
+	map[[StringMapsSize1]byte]struct{},
+	map[[StringMapsSize2]byte]struct{},
+	map[[StringMapsSize3]byte]struct{},
+	map[[StringMapsSize4]byte]struct{},
+	map[[StringMapsSize5]byte]struct{},
+	map[[StringMapsSize6]byte]struct{}) {
+	return map[[StringMapsSize1]byte]struct{}{},
+		map[[StringMapsSize2]byte]struct{}{},
+		map[[StringMapsSize3]byte]struct{}{},
+		map[[StringMapsSize4]byte]struct{}{},
+		map[[StringMapsSize5]byte]struct{}{},
+		map[[StringMapsSize6]byte]struct{}{}
+}
+
+func (k *KernelSelectorState) insertStringMaps(
+	stringMap1 map[[StringMapsSize1]byte]struct{},
+	stringMap2 map[[StringMapsSize2]byte]struct{},
+	stringMap3 map[[StringMapsSize3]byte]struct{},
+	stringMap4 map[[StringMapsSize4]byte]struct{},
+	stringMap5 map[[StringMapsSize5]byte]struct{},
+	stringMap6 map[[StringMapsSize6]byte]struct{}) [StringMapsNumSubMaps]uint32 {
+
+	details := [StringMapsNumSubMaps]uint32{}
+	mapid := uint32(0)
+	if len(stringMap1) > 0 {
+		mapid = uint32(len(k.stringMaps1))
+		k.stringMaps1 = append(k.stringMaps1, stringMap1)
+	} else {
+		mapid = 0xffffffff
+	}
+	details[0] = mapid
+
+	if len(stringMap2) > 0 {
+		mapid = uint32(len(k.stringMaps2))
+		k.stringMaps2 = append(k.stringMaps2, stringMap2)
```

Comment:
could please put in comment on how mapid (as size of the k.stringMapsX) correlates to the map_idx value logic in filter_char_buf_equal

Reply:
Yes I can do that.